lib/runner.js

```diff
@@ -507,8 +507,12 @@ Runner.prototype.runTests = function(suite, fn) {
       return;
     }
 
+    function parentPending(suite) {
+      return suite.pending || (suite.parent && parentPending(suite.parent));
+    }
+
     // pending
-    if (test.pending) {
+    if (test.pending || parentPending(test.parent)) {
```

Comment:
Not obvious. "parentPending" name means we testing parent, not item itself. I'd leave as is - more readable.